diff --git a/workspace/my-node-backend/src/db/database.ts b/workspace/my-node-backend/src/db/database.ts
index d2c1d5b..676b3a1 100644
--- a/workspace/my-node-backend/src/db/database.ts
+++ b/workspace/my-node-backend/src/db/database.ts
@@ -176,22 +176,28 @@ export const addTransaction = async (
 export const deleteTransaction = async (
     transaction_id: number
 ): Promise<ResultSetHeader> => {
-
-  await pool.query(
-    'DELETE FROM attachment WHERE transaction_id = ?',
+  const [result] = await pool.query<ResultSetHeader>(
+    `DELETE FROM transactions WHERE transaction_id = ?`,
     [transaction_id]
   );
 
+  return result;
+};
 
+export const deleteTransactionGoal = async (
+    goal_id: number
+): Promise<ResultSetHeader> => {
   const [result] = await pool.query<ResultSetHeader>(
-    `DELETE FROM transactions WHERE transaction_id = ?`,
-    [transaction_id]
+    `DELETE FROM transactions WHERE goal_id = ?`,
+    [goal_id]
   );
 
   return result;
 };
 
 
+
+
 export const recentTransactions = async (
     user_id: number,
 ): Promise<Transaction[]> => {
@@ -263,7 +269,7 @@ export const getGoals = async (
   user_id: number,
 ): Promise<Goal[]> => {
   const [rows] = await pool.query<Goal[]>(
-    `SELECT * FROM goals WHERE user_id = ?`,
+    `SELECT * FROM goals WHERE user_id = ? ORDER by priority ASC`,
     [user_id]
   );
 
diff --git a/workspace/my-node-backend/src/index.ts b/workspace/my-node-backend/src/index.ts
index 968d18b..02fe1e3 100644
--- a/workspace/my-node-backend/src/index.ts
+++ b/workspace/my-node-backend/src/index.ts
@@ -13,6 +13,7 @@ import notificationsRouter from "./routes/notifications.routes.js";
 import { fileURLToPath } from "url";
 import { getCategories } from "./db/database.js";
 
+
 dotenv.config();
 
 const app = express();
diff --git a/workspace/my-node-backend/src/routes/goals.routes.ts b/workspace/my-node-backend/src/routes/goals.routes.ts
index cc8257c..ca402f8 100644
--- a/workspace/my-node-backend/src/routes/goals.routes.ts
+++ b/workspace/my-node-backend/src/routes/goals.routes.ts
@@ -1,5 +1,5 @@
 import { Request, Response, NextFunction, Router } from "express";
-import { getGoals, addGoal, deleteGoal, getGoalCount, updateGoal, addNotification} from "../db/database.js";
+import { getGoals, addGoal, deleteGoal, getGoalCount, updateGoal, addNotification, deleteTransactionGoal} from "../db/database.js";
 
 const router = Router();
 
@@ -82,6 +82,9 @@ const deleteAGoal = async (
     }
 
     const { goal_id } = req.body;
+
+    await deleteTransactionGoal(goal_id);
+
     const result = await deleteGoal(Number(goal_id));
 
     return res.status(201).json({
diff --git a/workspace/my-node-backend/src/routes/transactions.routes.ts b/workspace/my-node-backend/src/routes/transactions.routes.ts
index 333caeb..d6fa4fc 100644
--- a/workspace/my-node-backend/src/routes/transactions.routes.ts
+++ b/workspace/my-node-backend/src/routes/transactions.routes.ts
@@ -62,7 +62,7 @@ const createTransaction = async (
     //console.log(result)
 
     if (goal_id) {
-      await updateGoalAmount(goal_id, amount, req.session.user.user_id);
+      await updateGoalAmount(req.session.user.user_id, Number(goal_id), Number(amount));
 
       await addNotification(
       req.session.user.user_id,
diff --git a/workspace/my-react-app/src/components/AppLayout.jsx b/workspace/my-react-app/src/components/AppLayout.jsx
index 5675c44..db5dc2c 100644
--- a/workspace/my-react-app/src/components/AppLayout.jsx
+++ b/workspace/my-react-app/src/components/AppLayout.jsx
@@ -1,12 +1,15 @@
 import { useNavigate, useLocation } from "react-router";
 import "../styles/AppLayout.css";
 
+const API_URL = import.meta.env.VITE_API_URL;
+
+
 export default function AppLayout({ user, children }) {
   const navigate = useNavigate();
   const location = useLocation();
 
   async function handleLogout() {
-    const res = await fetch("http://88.200.63.148:30040/api/logout", {
+    const res = await fetch(`${API_URL}/api/logout`, {
       method: "POST",
       credentials: "include",
     })
diff --git a/workspace/my-react-app/src/pages/AddBudgets.jsx b/workspace/my-react-app/src/pages/AddBudgets.jsx
index 5063b11..6fa1050 100644
--- a/workspace/my-react-app/src/pages/AddBudgets.jsx
+++ b/workspace/my-react-app/src/pages/AddBudgets.jsx
@@ -2,6 +2,7 @@ import { useEffect, useState } from "react";
 import { useNavigate, Link } from "react-router";
 import "../styles/AddBudgets.css"
 
+const API_URL = import.meta.env.VITE_API_URL;
 
 export default function AddBudgets() {
   const navigate = useNavigate();
@@ -16,7 +17,7 @@ export default function AddBudgets() {
 
     useEffect (() => {
         async function getCategories() {
-          const res2 = await fetch("http://88.200.63.148:30040/api/categories", {
+          const res2 = await fetch(`${API_URL}/api/categories`, {
           credentials: "include"
           });
   
@@ -34,7 +35,7 @@ export default function AddBudgets() {
     setMessage("");
 
     try {
-      const res = await fetch("http://88.200.63.148:30040/api/budgets/add", {
+      const res = await fetch(`${API_URL}/api/budgets/add`, {
         method: "POST",
         headers: {
           "Content-Type": "application/json",
diff --git a/workspace/my-react-app/src/pages/AddGoal.jsx b/workspace/my-react-app/src/pages/AddGoal.jsx
index 9a5e332..39547c7 100644
--- a/workspace/my-react-app/src/pages/AddGoal.jsx
+++ b/workspace/my-react-app/src/pages/AddGoal.jsx
@@ -2,6 +2,7 @@ import { useState } from "react";
 import { useNavigate, Link } from "react-router";
 import "../styles/Transactions.css"
 
+const API_URL = import.meta.env.VITE_API_URL;
 
 
 export default function AddGoal() {
@@ -20,7 +21,7 @@ export default function AddGoal() {
     setMessage("");
 
     try {
-      const res = await fetch("http://88.200.63.148:30040/api/goals/add", {
+      const res = await fetch(`${API_URL}/api/goals/add`, {
         method: "POST",
         headers: {
           "Content-Type": "application/json",
diff --git a/workspace/my-react-app/src/pages/AddTransaction.jsx b/workspace/my-react-app/src/pages/AddTransaction.jsx
index 153be85..eb4f139 100644
--- a/workspace/my-react-app/src/pages/AddTransaction.jsx
+++ b/workspace/my-react-app/src/pages/AddTransaction.jsx
@@ -2,6 +2,7 @@ import { useEffect, useState } from "react";
 import { useNavigate, Link, useLocation } from "react-router";
 import "../styles/AddTransactions.css"
 
+const API_URL = import.meta.env.VITE_API_URL;
 
 
 export default function AddTransaction() {
@@ -18,7 +19,7 @@ export default function AddTransaction() {
 
   useEffect (() => {
       async function getCategories() {
-        const res2 = await fetch("http://88.200.63.148:30040/api/categories", {
+        const res2 = await fetch(`${API_URL}/api/categories`, {
         credentials: "include"
         });
 
@@ -37,7 +38,7 @@ export default function AddTransaction() {
 
 
     try {
-      const res = await fetch("http://88.200.63.148:30040/api/transactions/add", {
+      const res = await fetch(`${API_URL}/api/transactions/add`, {
         method: "POST",
         headers: {
           "Content-Type": "application/json",
diff --git a/workspace/my-react-app/src/pages/Budgets.jsx b/workspace/my-react-app/src/pages/Budgets.jsx
index b724ccd..be7a339 100644
--- a/workspace/my-react-app/src/pages/Budgets.jsx
+++ b/workspace/my-react-app/src/pages/Budgets.jsx
@@ -3,7 +3,8 @@ import { useNavigate, useLocation } from "react-router";
 import "../styles/Budgets.css"
 
 
-const API_URL = "http://88.200.63.148:30040";
+const API_URL = import.meta.env.VITE_API_URL;
+
 
 export default function Budgets({ user }) {
   const [budgets, setBudgets] = useState([]);
diff --git a/workspace/my-react-app/src/pages/Goals.jsx b/workspace/my-react-app/src/pages/Goals.jsx
index 6d6c517..97dd5a8 100644
--- a/workspace/my-react-app/src/pages/Goals.jsx
+++ b/workspace/my-react-app/src/pages/Goals.jsx
@@ -3,7 +3,8 @@ import { useNavigate, useLocation } from "react-router";
 import "../styles/Goals.css"
 
 
-const API_URL = "http://88.200.63.148:30040";
+const API_URL = import.meta.env.VITE_API_URL;
+
 
 export default function Goals({ user }) {
   const [goals, setGoals] = useState([]);
@@ -31,8 +32,9 @@ export default function Goals({ user }) {
   }, []);
 
   const deleteGoal = async (goal_id) => {
+    console.log(goal_id)
     try {
-      const res = await fetch("http://88.200.63.148:30040/api/goals/delete", {
+      const res = await fetch(`${API_URL}/api/goals/delete`, {
         method: "DELETE",
         headers: {
           "Content-Type": "application/json",
@@ -69,7 +71,7 @@ export default function Goals({ user }) {
   
   const updateStatus = async (goal_id, name, status) => {
     try {
-      const res = await fetch("http://88.200.63.148:30040/api/goals/update", {
+      const res = await fetch("${API_URL}/api/goals/update", {
         method: "POST",
         headers: {
           "Content-Type": "application/json",
@@ -128,9 +130,14 @@ export default function Goals({ user }) {
               })}
             </p>
 
-            <p>
-              Required pace: €{perMonth}/month over {dif} months remaining
-            </p>
+              
+            {item.target_amount > item.current_amount && (
+              <p>
+                Required pace: €{perMonth}/month over {dif} months remaining
+              </p>
+            )}
+
+            <p>Priority: {item.priority}</p>
 
             <select
               value={item.status}
diff --git a/workspace/my-react-app/src/pages/Home.jsx b/workspace/my-react-app/src/pages/Home.jsx
index a0a229f..99079e8 100644
--- a/workspace/my-react-app/src/pages/Home.jsx
+++ b/workspace/my-react-app/src/pages/Home.jsx
@@ -1,7 +1,8 @@
 import { useState, useEffect } from "react";
 import "../styles/Home.css";
 
-const API_URL = "http://88.200.63.148:30040";
+const API_URL = import.meta.env.VITE_API_URL;
+
 
 export default function Home({ user }) {
   const [sum, setSum] = useState(0);
diff --git a/workspace/my-react-app/src/pages/Login.jsx b/workspace/my-react-app/src/pages/Login.jsx
index 6b893ca..0b50af0 100644
--- a/workspace/my-react-app/src/pages/Login.jsx
+++ b/workspace/my-react-app/src/pages/Login.jsx
@@ -1,7 +1,7 @@
 import { useState } from "react";
 import { useNavigate, Link } from "react-router";
 
-const API_URL = "http://88.200.63.148:30040";
+const API_URL = import.meta.env.VITE_API_URL;
 
 export default function Login({ setUser }) {
   const [email, setUsername] = useState("");
@@ -15,6 +15,7 @@ export default function Login({ setUser }) {
     setMessage("");
 
     try {
+      console.log(API_URL)
       const res = await fetch(`${API_URL}/api/users/login`, {
         method: "POST",
         headers: {
diff --git a/workspace/my-react-app/src/pages/Notifications.jsx b/workspace/my-react-app/src/pages/Notifications.jsx
index 0c29b1a..629a062 100644
--- a/workspace/my-react-app/src/pages/Notifications.jsx
+++ b/workspace/my-react-app/src/pages/Notifications.jsx
@@ -3,7 +3,8 @@ import { useNavigate, useLocation } from "react-router";
 import "../styles/Notifications.css"
 
 
-const API_URL = "http://88.200.63.148:30040";
+const API_URL = import.meta.env.VITE_API_URL;
+
 
 export default function Notifications({ user }) {
   const [notifications, setNotifications] = useState([]);
@@ -33,7 +34,7 @@ export default function Notifications({ user }) {
 
   const readAll = async () => {
      try {
-      const res = await fetch("http://88.200.63.148:30040/api/notifications/readAll", {
+      const res = await fetch(`${API_URL}/api/notifications/readAll`, {
         method: "POST",
         headers: {
           "Content-Type": "application/json",
diff --git a/workspace/my-react-app/src/pages/Register.jsx b/workspace/my-react-app/src/pages/Register.jsx
index 3e2ef8a..9ab6037 100644
--- a/workspace/my-react-app/src/pages/Register.jsx
+++ b/workspace/my-react-app/src/pages/Register.jsx
@@ -1,6 +1,9 @@
 import { useState } from "react";
 import { useNavigate, Link } from "react-router";
 
+const API_URL = import.meta.env.VITE_API_URL;
+
+
 export default function Register({ setUser }) {
   const navigate = useNavigate();
 
@@ -16,7 +19,7 @@ export default function Register({ setUser }) {
     setMessage("");
 
     try {
-      const res = await fetch("http://88.200.63.148:30040/api/users/register", {
+      const res = await fetch(`${API_URL}/api/users/register`, {
         method: "POST",
         headers: {
           "Content-Type": "application/json",
diff --git a/workspace/my-react-app/src/pages/Settings.jsx b/workspace/my-react-app/src/pages/Settings.jsx
index 799cc34..1238b9f 100644
--- a/workspace/my-react-app/src/pages/Settings.jsx
+++ b/workspace/my-react-app/src/pages/Settings.jsx
@@ -2,6 +2,9 @@ import { useEffect, useState } from "react";
 import { useNavigate, useLocation } from "react-router";
 import "../styles/Settings.css"
 
+const API_URL = import.meta.env.VITE_API_URL;
+
+
 
 export default function Settings({ user }) {
   const [first_name, setFirstName] = useState(user.first_name);
@@ -20,7 +23,7 @@ export default function Settings({ user }) {
     setMessage("");
 
     try {
-      const res = await fetch("http://88.200.63.148:30040/api/users/update", {
+      const res = await fetch(`${API_URL}/api/users/update`, {
         method: "POST",
         headers: {
           "Content-Type": "application/json",
@@ -52,7 +55,7 @@ export default function Settings({ user }) {
   };
 
   async function handleLogout() {
-    const res = await fetch("http://88.200.63.148:30040/api/logout", {
+    const res = await fetch(`${API_URL}/api/logout`, {
       method: "POST",
       credentials: "include",
     })
diff --git a/workspace/my-react-app/src/pages/Transactions.jsx b/workspace/my-react-app/src/pages/Transactions.jsx
index 8c930e7..ebbf6c1 100644
--- a/workspace/my-react-app/src/pages/Transactions.jsx
+++ b/workspace/my-react-app/src/pages/Transactions.jsx
@@ -2,7 +2,8 @@ import { useEffect, useState } from "react";
 import { useNavigate, useLocation } from "react-router";
 import "../styles/Transactions.css"
 
-const API_URL = "http://88.200.63.148:30040";
+const API_URL = import.meta.env.VITE_API_URL;
+
 
 export default function Transactions({ user }) {
   const [transactions, setTransactions] = useState([]);
@@ -34,7 +35,7 @@ export default function Transactions({ user }) {
 
   const deleteTransaction = async (transaction_id) => {
     try {
-      const res = await fetch("http://88.200.63.148:30040/api/transactions/delete", {
+      const res = await fetch(`${API_URL}/api/transactions/delete`, {
         method: "DELETE",
         headers: {
           "Content-Type": "application/json",
@@ -68,7 +69,7 @@ export default function Transactions({ user }) {
     formData.append("transaction_id", transaction_id);
 
     try {
-      const res = await fetch("http://88.200.63.148:30040/api/transactions/upload", {
+      const res = await fetch(`${API_URL}/api/transactions/upload`, {
         method: "POST",
         credentials: "include",
         body: formData,
@@ -90,7 +91,7 @@ export default function Transactions({ user }) {
 
   const deleteFile = async (attachment_id) => {
     try {
-      const res = await fetch("http://88.200.63.148:30040/api/transactions/deleteFile", {
+      const res = await fetch(`${API_URL}/api/transactions/deleteFile`, {
         method: "DELETE",
         headers: {
           "Content-Type": "application/json",
diff --git a/workspace/my-react-app/src/routes/AppRouter.jsx b/workspace/my-react-app/src/routes/AppRouter.jsx
index c6c3943..6194f40 100644
--- a/workspace/my-react-app/src/routes/AppRouter.jsx
+++ b/workspace/my-react-app/src/routes/AppRouter.jsx
@@ -3,6 +3,8 @@ import { useState, useEffect } from "react";
 
 //  import "../styles/App.css"
 
+const API_URL = import.meta.env.VITE_API_URL;
+
 import LandingPage from "../LandingPage";
 import Login from "../pages/Login";
 import Register from "../pages/Register";
@@ -26,7 +28,7 @@ export default function App() {
   useEffect(() => {
     async function fetchUser() {
       try {
-        const res = await fetch("http://88.200.63.148:30040/api/me", {
+        const res = await fetch(`${API_URL}/api/me`, {
           credentials: "include",
         });
         
diff --git a/workspace/my-react-app/src/styles/Home.css b/workspace/my-react-app/src/styles/Home.css
index 8c4aca1..8f3a22a 100644
--- a/workspace/my-react-app/src/styles/Home.css
+++ b/workspace/my-react-app/src/styles/Home.css
@@ -42,7 +42,12 @@
 
 .home-recent,
 .spent-by-category {
-  width: 50%;
+  width: 50%; 
+  background-color: #011538;
+  border-radius: 10px;
+  padding: 10px;
+  border: 1px solid #ccc;
+  border-radius: 8px;
 }
 
 .recent-list {
diff --git a/workspace/my-react-app/src/styles/Transactions.css b/workspace/my-react-app/src/styles/Transactions.css
index fbcf771..3d03626 100644
--- a/workspace/my-react-app/src/styles/Transactions.css
+++ b/workspace/my-react-app/src/styles/Transactions.css
@@ -42,44 +42,4 @@
     opacity: 0.9;
 }
 
-/* TRANSACTIONS LIST */
-.recent {
-  display: flex;
-  flex-direction: column;
-  gap: 10px;
-}
-
-.recent-item {
-  display: flex;
-  justify-content: space-between;
-  padding: 12px;
-  border: 1px solid #ccc;
-  border-radius: 8px;
-  background: #011538;
-}
-
-.recent-item .desc {
-  margin: 0;
-  font-weight: 500;
-}
-
-.recent-item .meta {
-  margin: 4px 0 0 0;
-  opacity: 0.7;
-  font-size: 12px;
-}
-
-.recent-item .right {
-  text-align: right;
-}
-
-.recent-item .amount {
-  margin: 0;
-  font-weight: bold;
-}
 
-.recent-item .date {
-  margin: 4px 0 0 0;
-  font-size: 12px;
-  opacity: 0.7;
-}
\ No newline at end of file
